src/components/Editor/parsers/parse.ts

```diff
@@ -8,11 +8,26 @@ export const parsers = {
   'cses.fi': parseCses,
   'atcoder.jp': parseAc,
 };
-export default function parse(url: string, html: string) {
+export async function getHtml(url: string): Promise<string> {
+  const res = await fetch('/api/fetch-html', {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+    },
+    body: JSON.stringify({ url }),
+  }).then(res => res.json());
+  return res.data;
```

Comment:
I feel like the logic of parsing the HTML page belong on the server, not the client. It feels a bit weird to just have an `/api/fetch-html` endpoint like this. I'm fine with leaving it as-is though because it seems like a lot of effort to change it and I don't think it's a big deal